play-json/shared/src/main/scala/JsonConfiguration.scala

```diff
@@ -21,16 +21,25 @@ object JsonConfiguration {
     type Opts = O
   }
 
+  // This exists for binary compatibility. It's protected to avoid ambiguous
+  // argument compile errors with the other apply, but Scala protected methods
+  // compile to public Java methods, so from a binary perspective it's fine.
+  protected def apply(
+    naming: JsonNaming
+  ): JsonConfiguration.Aux[Json.MacroOptions] = new Impl(naming)
+
   /**
-   * @tparam O the options for the JSON macros
    * @param naming the naming strategy
    */
-  def apply[O <: Json.MacroOptions](
+  def apply[Opts <: Json.MacroOptions: Json.DefaultMacroOptions](
```

Comment:
Previously, if you didn't supply a type, you'd end up with `JsonConfiguration.Aux[Nothing]`, and since `Nothing` is a bottom type, ie, it's a sub class of everything, that meant every option was enabled. So `WithDefaultValues` has been enabled by default - that's a bug. I think.